From 352d797e69ffe630e88e181f042343332d895e94 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 12 Jul 2016 14:48:40 +0800
Subject: [PATCH 1111/5242] MLK-12980-3 video: mxsfb: free 'pseudo_palette' to
 avoid memory leak

commit  8548207e68281a3e600184a8138be2f794039ebd from
https://source.codeaurora.org/external/imx/linux-imx.git

The dynamicly allocated 'pseudo_palette' area should be
freed when unused to avoid memory leak.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 2da4196..0c23932 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1510,6 +1510,7 @@ static int mxsfb_probe(struct platform_device *pdev)
 	fb_destroy_modelist(&fb_info->modelist);
 fb_pm_runtime_disable:
 	pm_runtime_disable(&host->pdev->dev);
+	devm_kfree(&pdev->dev, fb_info->pseudo_palette);
 fb_release:
 	framebuffer_release(fb_info);
 
@@ -1528,6 +1529,7 @@ static int mxsfb_remove(struct platform_device *pdev)
 	unregister_framebuffer(fb_info);
 	mxsfb_free_videomem(fb_info);
 
+	devm_kfree(&pdev->dev, fb_info->pseudo_palette);
 	framebuffer_release(fb_info);
 
 	platform_set_drvdata(pdev, NULL);
-- 
1.7.9.5

